### PR TITLE
RSDK-3550: Fully modularize the cartographer module

### PR DESCRIPTION
**Pull request by nicksanford**
# Description

[Ticket](https://viam.atlassian.net/browse/RSDK-3550)

NOTE: This will be merged once cartographer-module `v0.3.20-rc0` is promoted to stable

This makes the cartographer docs changes required by the Fully Modularize Cartographer project.

They include:
1. Update docs to say that sensors list may not be empty (already done previously)
2. Remove all references to delete_process_data, use_live_data, port
3. Mark data_rate_msec as deprecated in [Viam Documentation](http://docs.viam.com/)

### CODE REVIEW

File: docs/services/slam/cartographer/_index.md [hunk 1]
```diff
@@ -281,9 +276,7 @@ Watch a map start to appear.
 | `data_dir` | string | **Required** | Path to [the directory](#mapping-modes) used for saving input LIDAR data in <file>/data</file> and output mapping data in <file>/map</file>. |
 | `sensors` | string[] | **Required** | Names of configured RPlidar devices providing data to the SLAM service. |
 | `map_rate_sec` | int | Optional | Rate of <file>/map</file> generation *(seconds)*. <ul> Default: `60`. </ul> |
-| `data_rate_msec` | int | Optional | Rate of <file>/data</file> collection from `sensors` *(milliseconds)*. <ul>Default: `200`.</ul> |
-| `port` | string | Optional | Port for SLAM gRPC server. If running locally, this should be in the form "localhost:<PORT>". If no value is specified a random available port is assigned. |
-| `delete_processed_data` | bool | Optional | <p>Setting this to `true` helps to reduce the amount of memory required to run SLAM.</p> <ul> `true`: sensor data is deleted after the SLAM algorithm has processed it. </ul><ul> `false`: sensor data is not deleted after the SLAM algorithm has processed it. </ul> |
+| `data_rate_msec` | int | Deprecated | Rate of <file>/data</file> collection from `sensors` *(milliseconds)*. <ul>Default: `200`.</ul> |
```
Comment (nicksanford) [inline docs/services/slam/cartographer/_index.md:279]: @npentrel is this the proper way to mark a config parameter as deprecated?

cc @EshaMaharishi

**Review by kim-mishra — APPROVED** [review]
LGTM

File: docs/services/slam/cartographer/_index.md (hunk 1 above)
Comment (npentrel) [reply docs/services/slam/cartographer/_index.md:279]: Oh - I think this is the first time we're encountering this. Does it still work? If so this is ok. Will need to think about how we generally handle this but for now this is fine to merge

**Review by npentrel — APPROVED** [review]

File: docs/services/slam/cartographer/_index.md (hunk 1 above)
Comment (nicksanford) [reply docs/services/slam/cartographer/_index.md:279]: It does still work, but is planned on being removed as part of a future project.

**Comment by npentrel** [comment]
@nicksanford is this ready to merge?

**Comment by nicksanford** [comment]
@npentrel not quite yet, the software change this describes hasn't been released yet. 

I created this PR so the docs would be ready to go.

I'll give you a heads up before I merge it.

**Comment by nicksanford** [comment]
@npentrel merging this in a few min now that there is a new release of cartographer-module out: https://github.com/viamrobotics/viam-cartographer/releases/tag/v0.3.22

**Pull request merged**